src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Url/Base64UrlDecoder.cs

```diff
@@ -0,0 +1,794 @@
+        /// </exception>
+        public static int GetMaxDecodedLength(int base64Length)
+        {
+            ArgumentOutOfRangeException.ThrowIfNegative(base64Length);
+
+            (uint whole, uint remainder) = uint.DivRem((uint)base64Length, 4);
+
+            return (int)(whole * 3 + (remainder > 0 ? remainder - 1 : 0));
+        }
+
+        /// <summary>
+        /// Decode the span of UTF-8 encoded text represented as Base64Url into binary data.
+        /// </summary>
+        /// <param name="source">The input span which contains UTF-8 encoded text in Base64Url that needs to be decoded.</param>
+        /// <param name="destination">The output span which contains the result of the operation, i.e. the decoded binary data.</param>
+        /// <param name="bytesConsumed">The number of input bytes consumed during the operation. This can be used to slice the input for subsequent calls, if necessary.</param>
+        /// <param name="bytesWritten">The number of bytes written into the output span. This can be used to slice the output for subsequent calls, if necessary.</param>
+        /// <param name="isFinalBlock"><see langword="true"/> (default) when the input span contains the entire data to encode.
+        /// Set to <see langword="true"/> when the source buffer contains the entirety of the data to encode.
+        /// Set to <see langword="false"/> if this method is being called in a loop and if more input data may follow.
+        /// At the end of the loop, call this (potentially with an empty source buffer) passing <see langword="true"/>.</param>
+        /// <returns>It returns the OperationStatus enum values:
+        /// - Done - on successful processing of the entire input span
+        /// - DestinationTooSmall - if there is not enough space in the output span to fit the decoded input
+        /// - NeedMoreData - only if <paramref name="isFinalBlock"/> is false and the input is not a multiple of 4
+        /// - InvalidData - if the input contains bytes outside of the expected Base64Url range, or if it contains invalid/more than two padding characters
+        /// or if the input is incomplete (i.e. the remainder of <paramref name="source"/> % 4 is 1) and <paramref name="isFinalBlock"/> is <see langword="true"/>.
+        /// </returns>
+        /// <remarks>
+        /// As padding is optional the <paramref name="source"/> length not required to be a multiple of 4 even if <paramref name="isFinalBlock"/> is <see langword="true"/>.
+        /// If the <paramref name="source"/> length is not a multiple of 4 and <paramref name="isFinalBlock"/> is <see langword="true"/> the remainders decoded accordingly:
+        /// Remainder of 3 bytes - decoded into 2 bytes data, decoding succeeds.
+        /// Remainder of 2 bytes - decoded into 1 byte data. decoding succeeds.
+        /// Remainder of 1 byte - will cause OperationStatus.InvalidData result.
```

Comment:
I think this will just be 4 sentences strung together on one line.  If newlines are important here then `<br />` might help (not sure if that flows all the way through to the docs site).  It's probably best represented as a table, though I don't know what syntax to use here to flow through to docs as a table.

Reply:
I prefer separate lines, don't see any instruction about addong a table in https://github.com/dotnet/dotnet-api-docs/wiki, though I think a list would work for keeping the rows as is.